docs/kap_proposals/kap_8_google_secret_management.md

```diff
@@ -0,0 +1,83 @@
+# Support for [Google Secret Manager](https://cloud.google.com/secret-manager/)
+
+This feature will enable users to retrieve secrets from Google Secret Manager API using the `gsm` keyword.
+
+## Specification
+
+`project_id` uniquely identifies GCP projects, and it needs to be made accessible to kapitan in one of the following ways:
+
+- As a part of target
+```yaml
+parameters:
+  kapitan:
+    secrets:
+      gsm:
+        project_id: Project_Id
+```
+
+- As a flag
+```shell
+$ kapitan refs --google-project-id=<Project_Id> --write gsm:/path/to/secret_id -f secret_id_file.txt
+```
+
+- As an environment variable
+```shell
+export PROJECT_ID=<Project_Id>
+```
+
+## Using a secret
+
+`secret_id` is used to point to the secret data. The following command will be used to add a `secret_id` to kapitan.
```

Comment:
Can you add a small explanation of what secret_id is in the context of GSM? Just to make things clearer as this is a bit different from other secret backends. :)